tests/regression/test_aw.py

```diff
@@ -0,0 +1,123 @@
+@pytest.fixture(scope='module', params=["conforming", "nonconforming"])
+def stress_element(request):
+    if request.param == "conforming":
+        return FiniteElement("AWc", triangle, 3)
+    elif request.param == "nonconforming":
+        return FiniteElement("AWnc", triangle, 2)
+    else:
+        raise ValueError("Unknown family")
+
+
+def test_aw(stress_element):
+    N_base = 2
+    mesh = UnitSquareMesh(N_base, N_base)
+    mh = MeshHierarchy(mesh, 4)
+
+    V = FunctionSpace(mesh, mesh.coordinates.ufl_element())
+
+    # Warp the meshes
+    eps = Constant(1 / 2**(N_base-1))
+    x, y = SpatialCoordinate(mesh)
+    new = Function(V).interpolate(as_vector([x + eps*sin(2*pi*x)*sin(2*pi*y),
+                                             y - eps*sin(2*pi*x)*sin(2*pi*y)]))
+    coords = [new]
+    for mesh in mh[1:]:
+        fine = Function(mesh.coordinates.function_space())
+        prolong(new, fine)
+        coords.append(fine)
+        new = fine
+    for mesh, coord in zip(mh, coords):
+        mesh.coordinates.assign(coord)
+
+    nu = Constant(0.25)
+    lam = Constant(1)
+    mu = lam*(1 - 2*nu)/(2*nu)
+
+    I = Identity(2)
+
+    # Evaluation of a constant compliance tensor
+    # (in the homogeneous isotropic case)
+    def A(sig):
+        return (1/(2*mu))*(sig - nu*tr(sig)*I)
+
+    # Linearised strain rate tensor
+    def epsilon(u):
+        return sym(grad(u))
+
+    l2_u = []
+    l2_sigma = []
+    l2_div_sigma = []
+
+    element = MixedElement([stress_element, VectorElement("DG", mesh.ufl_cell(), 1)])
+    for msh in mh[1:]:
+        x, y = SpatialCoordinate(msh)
+        uex = as_vector([sin(pi*x)*sin(pi*y), sin(pi*x)*sin(pi*y)])
+        sigex = as_tensor([[cos(pi*x)*cos(3*pi*y), y + 2*cos(pi*x/2)],
+                           [y + 2*cos(pi*x/2), -sin(3*pi*x)*cos(2*pi*x)]])
+
+        # Compliance and constraint residuals of the MMS
+        comp_r = A(sigex) - epsilon(uex)
+        cons_r = div(sigex)
+
+        V = FunctionSpace(msh, element)
+
+        Uh = Function(V)
+
+        (sigh, uh) = split(Uh)
+        (tau, v) = TestFunctions(V)
+
+        n = FacetNormal(msh)
+
+        # Hellinger--Reissner residual, incorporating MMS residuals
+        F = (
+            + inner(A(sigh), tau)*dx
+            + inner(div(tau), uh)*dx
+            + inner(div(sigh), v)*dx
+            - inner(comp_r, tau)*dx
+            - inner(cons_r, v)*dx
+            - inner(dot(tau, n), uex)*ds
+            )  # noqa: E123
+
+        params = {"snes_type": "newtonls",
+                  "snes_linesearch_type": "basic",
+                  "snes_max_it": 100,
+                  "snes_monitor": None,
+                  "snes_converged_reason": None,
+                  "ksp_type": "preonly",
+                  "pc_type": "lu",
+                  "pc_factor_mat_solver_type": "umfpack",
+                  "mat_type": "aij",
+                  "snes_rtol": 1e-5,
+                  "snes_atol": 1e-25,
+                  "snes_stol": 0.0}
```

Comment:
do we know how essential mumps and/or iterative refinement are to the example, or can we get away with more vanilla options?

Reply:
`snes_type`: `ksponly` works. Thanks for the suggestion. We need some LU library not PETSc because the problem is of saddle point type, but both UMFPACK and MUMPS will work.

Reply:
snes was for iterative refinement, if you turn that to ksponly, you can also remove all the other snes-related options/tolerances

Reply:
I got rid of all the tolerances.